src/aux-records/AIController.ts

```diff
@@ -598,6 +610,25 @@ export class AIController {
                 };
             }
 
+            if (allowedFeatures.ai.chat.allowedModels) {
+                const allowedModels = allowedFeatures.ai.chat.allowedModels;
+                if (
+                    !allowedModels ||
+                    allowedModels.length === 0 ||
+                    allowedModels.includes(model)
+                ) {
+                    return {
+                        success: true,
+                    };
+                }
```

Comment:
@TroyceGowdy This still looks wrong. In this case, if an empty list of allowed models is specified, then a success result is returned to the user, when in fact an empty list of allowed models should always cause the request to be not authorized.

You should be able to just remove that if statement and return to fix the issue.